src/components/Navigation/NavLink.component.tsx

```diff
@@ -0,0 +1,31 @@
+import styled from '@emotion/styled';
+import { Theme } from '@mui/material';
+import {
+  NavLink as NavLinkBase,
+  NavLinkProps as NavLinkBaseProps,
+} from 'react-router-dom';
+
+type NavLinkProps = NavLinkBaseProps & {
+  theme?: Theme;
+};
+
+export const NavLink = styled(NavLinkBase)(({ theme }: NavLinkProps) => ({
```

Comment:
```suggestion
export const NavLink = styled(NavLinkBase)(({ theme }) => ({
```
and you don't have to extend `NavLinkProps` with `theme`